### PR TITLE
Fix gengamma _nump failure

### PR DESCRIPTION
**Pull request by ldamewood**
Issue #4724

### CODE REVIEW

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -2039,7 +2039,7 @@ def _ppf(self, q, a, c):
         return where(cond > 0, val1**ic, val2**ic)
 
     def _munp(self, n, a, c):
-        return special.gamma(a+n*1.0/c) / special.gamma(a)
+        return special.poch(a,n*1.0/c)
```
Comment (josef-pkt) [inline scipy/stats/_continuous_distns.py]: style cleanup can be included, pep-8 
spaces around operators

**Comment by josef-pkt** [comment]
looks good to me

worth a check: adding this parameters (temporarily) to the list of distributions would check whether all other methods work for this large parameter values

**Comment by josef-pkt** [comment]
can you add a code comment about `poch`,  with the old formula or link to isssue
I didn't know anything about `poch`

**Comment by argriffing** [comment]
I don't see `poch` at http://docs.scipy.org/doc/scipy/reference/special.html or http://docs.scipy.org/doc/scipy-dev/reference/special.html.

**Comment by ldamewood** [comment]
`poch(n,c)` is some C code and should be identical to the division of two gamma functions, which the original code had, but it should not have the same overflow problems. I don't know why it is undocumented, but it is called the Pochhammer symbol according to the C comments https://en.wikipedia.org/wiki/Pochhammer_symbol. I hope the new one is clearer.

File: scipy/stats/_continuous_distns.py [hunk 2]
```diff
@@ -2039,7 +2039,10 @@ def _ppf(self, q, a, c):
         return where(cond > 0, val1**ic, val2**ic)
 
     def _munp(self, n, a, c):
-        return special.gamma(a+n*1.0/c) / special.gamma(a)
+        # return special.gamma(a+n*1.0/c) / special.gamma(a)
```
Comment (WarrenWeckesser) [inline scipy/stats/_continuous_distns.py]: Remove this line; there is no need to keep code that has been commented-out.  The comment below is a sufficient reminder of the meaning of `poch`.

**Comment by ev-br** [comment]
The change to `poch` looks good. Merged in 
 https://github.com/scipy/scipy/commit/4f65e8ec31026a8b6b6c6a18c4ea23a97f0bcc4a, thanks @ldamewood, all.

**Comment by ev-br** [comment]
@ldamewood I took liberty to squash your three commits and rewrite the commit message in the format recommended at http://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html

You are the author of the relevant git commit, https://github.com/scipy/scipy/commit/a7ded12a17a50df8a4ce7f1cca5ee6fdfd080f22, your name will be included into the release notes for the upcoming release. Thanks for the fix, and keep them coming :-).

**Comment by ev-br** [comment]
@josef-pkt I ran the tests for gengamma with these parameters. It looks like `logcdf` is problematic at these parameters: it just computes `log(_cdf)`, and `_cdf` is just `special.gammainc`, which underflows. Can probably be fixed, not sure if it's worth it.

**Comment by josef-pkt** [comment]
@ev-br  Thanks, I don't know how relevant the case for large parameters is.
We should have a place where we can store this kind of information, known numerical limitations of the current implementation. Maybe at some point someone runs into the problem and the solution to improve it.

**Pull request closed without merging**